flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -159,34 +161,6 @@ public void testRestartAutomatically() throws Exception {
 	}
 
 	@Test
-	public void taskShouldFailWhenFailureRateLimitExceeded() throws Exception {
```

Comment:
Yes, as part of introducing the "callback" indirection, we can now also test the restart strategies much better, without always setting up a full ExecutionGraph. I added it to the refactoring.